lib/client.js

```diff
@@ -44,10 +45,14 @@ export default class Client {
         }
 
         let data;
-        await fetch(fullUrl, options).then(response => {
+        await fetch(fullUrl, options).then(async (response) => {
             if (response.ok) {
                 if (method === 'get') {
-                    data = response.text();
+                    data = await response.text();
+
+                    if (isJSON(data)) {
```

Comment:
I think it should be better to use the header of response : `response.headers.get('content-type')` and checking if it contains `application/json`. It will avoid using `JSON.parse`